server/plugin/plugin.go

```diff
@@ -89,6 +89,10 @@ func (p *MatterpollPlugin) OnActivate() error {
 		return errors.Wrap(err, "failed to patch bot description")
 	}
 
+	if err := p.API.RegisterCommand(p.getCommand(p.getConfiguration().Trigger)); err != nil {
```

Comment:
Should we instead be calling `OnConfigurationChange()` here? Until that's called, `p.getConfiguration()` will just be returning the nil value for the `configuration` struct, which seems like it won't yet have the configured `trigger`.

Reply:
@lieut-data's suggestion looks good to me. But there is a concern that `OnConfigurationChange()` relies on `p.isActivated()`, and `p.isActivated()` does not return `true` until `p.setActivated(true)` in few lines later is done. So I think that `OnConfigurationChange()` should be called after `p.setActivated(true)`. (but it looks little strange)

Reply:
The MM server ensures that `OnConfigurationChange` is called before `OnActivate` https://github.com/mattermost/mattermost-server/blob/57244d06393502eba388bf8c140b8158cc3da80a/plugin/client_rpc.go#L214. I'm inclined to rely on that mechanism.

Reply:
Thanks @hanzei. I missed that mechanism has been documented.  https://developers.mattermost.com/extend/plugins/server/reference/#Hooks.OnConfigurationChange. So I'm fine with your changes.